src/renderer/components/annotate-area.tsx

```diff
@@ -0,0 +1,229 @@
+import { LazyBrush } from 'lazy-brush';
+import * as React from 'react';
+import { IImageDimensions, IPath, IPoint, Tool } from './snipping-tool';
+
+const { useState } = React;
+
+export interface ISvgPath {
+  svgPath: string;
+  key: string;
+  strokeWidth: number;
+  color: string;
+  shouldShow: boolean;
+}
+
+export interface IAnnotateAreaProps {
+  paths: IPath[];
+  highlightColor: string;
+  penColor: string;
+  onChange: (paths: IPath[]) => void;
+  imageDimensions: IImageDimensions;
+  chosenTool: Tool;
+  screenSnippetPath: string;
+}
+
+const lazy = new LazyBrush({
+  radius: 3,
+  enabled: true,
+  initialPoint: { x: 0, y: 0 },
+});
+const TOP_MENU_HEIGHT = 48;
+const MIN_ANNOTATE_AREA_HEIGHT = 200;
+const MIN_ANNOTATE_AREA_WIDTH = 312;
+const PEN_WIDTH = 5;
+const HIGHLIGHT_WIDTH = 28;
+
+const AnnotateArea: React.FunctionComponent<IAnnotateAreaProps> = ({ paths, highlightColor, penColor, onChange, imageDimensions, chosenTool, screenSnippetPath }) => {
+  const [isDrawing, setIsDrawing] = useState(false);
+
+  const maybeErasePath = (key: string) => {
+    // erase logic here
+    return key;
+  };
+
+  const stopDrawing = () => {
+    if (isDrawing) {
+      setIsDrawing(false);
+    }
+  };
+
+  // Utility functions
+
+  const getMousePosition = (e: React.MouseEvent) => {
+    // We need to offset for elements in the window that is not the annotate area
+    const x = imageDimensions.width >= MIN_ANNOTATE_AREA_WIDTH ? e.pageX : e.pageX - (MIN_ANNOTATE_AREA_WIDTH - imageDimensions.width) / 2;
+    const y = imageDimensions.height >= MIN_ANNOTATE_AREA_HEIGHT ? (e.pageY - TOP_MENU_HEIGHT) : (e.pageY - ((MIN_ANNOTATE_AREA_HEIGHT - imageDimensions.height) / 2) - TOP_MENU_HEIGHT);
+    return { x, y };
+  };
+
+  // Render and preparing render functions
+
+  const addHighlightPoint = (paths: IPath[], point: IPoint) => {
+    const activePath = paths[paths.length - 1];
+    const shouldShow = true;
+    const key = 'path' + paths.length;
+    if (!isDrawing) {
+      paths.push({
+        points: [point],
+        color: highlightColor,
+        strokeWidth: HIGHLIGHT_WIDTH,
+        shouldShow,
+        key,
+      });
+    } else {
+      activePath.points.push(point);
```

Comment:
Do you want to consider early return if condition? 😄 It's just a coding style anyway.

Reply:
I can swap the if condition and therefore make it a bit more readable. Early return I don't think suits this use case, since the return is the same for both conditions.